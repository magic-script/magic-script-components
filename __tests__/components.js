// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';

import {
    Audio,
    Model,
    Video,
    Quad,
    Spinner,
    ListView,
    DropdownList,
    RectLayout,
    Dialog,
    ProgressBar,
    Button,
    Text
} from '../src/components/components.js';

test('Text component', () => {
  const localPosition = [1, 1, 1];
  const localRotation = [1, 1, 1, 1];
  const localScale= [1, 1, 1];
  const localTransform= [1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4];
  const alignment = 'top-left';
  const charSpacing = 0.01;
  const textAlignment = 'center';

  const boundsSize = {
    boundsSize: [0.6, 0.4],
    wrap: true
  };

  const fontParameters = {
    fontStyle: 'normal',
    fontWeight: 'regular',
    fontSize: 0.05,
    allCaps: true
  };

  const component = renderer.create(
    <Text
      position={localPosition}
      rotation={localRotation}
      scale={localScale}
      transform={localTransform}
      anchorPoint={alignment}
      letterSpacing={charSpacing}
      textAlign={textAlignment}
      width={boundsSize.boundsSize[0]}
      height={boundsSize.boundsSize[1]}
      multiline={boundsSize.wrap}
      fontStyle={fontParameters.fontStyle}
      fontWeight={fontParameters.fontWeight}
      fontSize={fontParameters.fontSize}
      allCaps={fontParameters.allCaps}
    >MagicScript Components</Text>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(component).toHaveProperty('localPosition', localPosition);
  expect(tree).toHaveProperty('localRotation', localRotation);
  expect(tree).toHaveProperty('localScale', localScale);
  expect(tree).toHaveProperty('localTransform', localTransform);
  expect(tree).toHaveProperty('alignment', alignment);
  expect(tree).toHaveProperty('charSpacing', charSpacing);
  expect(tree).toHaveProperty('textAlignment', textAlignment);
  expect(tree).toHaveProperty('boundsSize.wrap', true);
  expect(tree).toHaveProperty('boundsSize.boundsSize', [0.6, 0.4]);
  expect(tree).toHaveProperty('fontParameters.fontStyle', fontParameters.fontStyle);
  expect(tree).toHaveProperty('fontParameters.fontWeight', fontParameters.fontWeight);
  expect(tree).toHaveProperty('fontParameters.fontSize', fontParameters.fontSize);
  expect(tree).toHaveProperty('fontParameters.allCaps', fontParameters.allCaps);
});

test('Audio component', () => {
  const fileName = 'resources/audio.mp3';
  const soundLooping = true;
  const soundMute = false;
  const soundPitch = 1;
  const soundVolumeLinear = 1;
  const streamedFileOffset = 0.5;

  const component = renderer.create(
    <Audio
      path={fileName}
      looping={soundLooping}
      mute={soundMute}
      pitch={soundPitch}
      volume={soundVolumeLinear}
      seekTo={streamedFileOffset}
    />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('fileName', fileName);
  expect(tree).toHaveProperty('soundLooping', soundLooping);
  expect(tree).toHaveProperty('soundMute', soundMute);
  expect(tree).toHaveProperty('soundPitch', soundPitch);
  expect(tree).toHaveProperty('soundVolumeLinear', soundVolumeLinear);
  expect(tree).toHaveProperty('streamedFileOffset', streamedFileOffset);
});

test('Video component', () => {
  const videoPath = 'resources/video.mp4';
  const timedTextPath = 'resources/subtitles.txt';

  const component = renderer.create(
    <Video
      path={videoPath}
      subtitlePath={timedTextPath}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('videoPath', videoPath);
  expect(tree).toHaveProperty('timedTextPath', timedTextPath);
});

test('Model component', () => {
  const modelPath = 'resources/model.glb';

  const component = renderer.create(
    <Model path={modelPath}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('modelPath', modelPath);
});

test('Quad component', () => {
  const size = [0.5, 0.4];

  const component = renderer.create(
    <Quad
      width={size[0]}
      height={size[1]}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('size', size);
});

test('Spinner component', () => {
  const size = [0.3, 0.3];

  const component = renderer.create(
    <Spinner
      width={size[0]}
      height={size[1]}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('size', size);
});

test('ListView component', () => {
  const scrollingEnabled = true;

  const component = renderer.create(
    <ListView scrollEnabled={scrollingEnabled}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('scrollingEnabled', scrollingEnabled);
});

test('DropdownList component', () => {
  const textSize = 0.08;
  const listTextSize = 0.05;

  const component = renderer.create(
    <DropdownList
      fontSize={textSize}
      listFontSize={listTextSize}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('textSize', textSize);
  expect(tree).toHaveProperty('listTextSize', listTextSize);
});

test('RectLayout component', () => {
  const contentAlignment = 'top-left';

  const component = renderer.create(
    <RectLayout alignment={contentAlignment}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('contentAlignment', contentAlignment);
});

test('Dialog component', () => {
  const scrolling = true;

  const component = renderer.create(
    <Dialog scrollEnabled={scrolling}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('scrolling', scrolling);
});

test('ProgressBar component', () => {
  const progressColor = {
    beginColor: 'white',
    endColor: 'blue'
  };

  const component = renderer.create(
    <ProgressBar
      beginColor={progressColor.beginColor}
      endColor={progressColor.endColor}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('progressColor.beginColor', progressColor.beginColor);
  expect(tree).toHaveProperty('progressColor.endColor', progressColor.endColor);
});

test('Button component', () => {
  const labelSide = 'left';

  const component = renderer.create(
    <Button textSide={labelSide}/>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveProperty('labelSide', labelSide);
});
