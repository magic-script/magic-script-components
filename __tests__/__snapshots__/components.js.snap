// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio component 1`] = `
<audio
  looping={true}
  mute={false}
  path="resources/audio.mp3"
  pitch={1}
  seekTo={0.5}
  volume={1}
/>
`;

exports[`Button component 1`] = `
<button
  alignment="center-center"
  debug={false}
  enabled={true}
  height={0}
  hideBounds={false}
  iconColor="white"
  iconSize={0.08}
  labelSide="left"
  localPosition={
    Array [
      0,
      0,
      0,
    ]
  }
  localRotation={
    Array [
      0,
      0,
      0,
      1,
    ]
  }
  localScale={
    Array [
      1,
      1,
      1,
    ]
  }
  localTransform={
    Array [
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      0,
    ]
  }
  skipRaycast={false}
  textColor="white"
  textSize={0.08}
  visible={true}
  width={0}
>

</button>
`;

exports[`Dialog component 1`] = `
<dialog
  scrollEnabled={true}
/>
`;

exports[`DropdownList component 1`] = `
<dropdownList
  fontSize={0.08}
  listFontSize={0.05}
/>
`;

exports[`ListView component 1`] = `
<listView
  scrollEnabled={true}
/>
`;

exports[`Model component 1`] = `
<model
  path="resources/model.glb"
/>
`;

exports[`ProgressBar component 1`] = `
<progressBar
  beginColor="white"
  endColor="blue"
/>
`;

exports[`Quad component 1`] = `
<quad
  height={0.4}
  width={0.5}
/>
`;

exports[`RectLayout component 1`] = `
<rectLayout
  alignment="top-left"
/>
`;

exports[`Spinner component 1`] = `
<spinner
  height={0.3}
  width={0.3}
/>
`;

exports[`Text component 1`] = `
<text
  alignment="top-left"
  allCaps={true}
  boundsSize={
    Object {
      "boundsSize": Array [
        0.6,
        0.4,
      ],
      "wrap": true,
    }
  }
  charSpacing={0.01}
  debug={false}
  enabled={true}
  fontParameters={
    Object {
      "allCaps": true,
      "fontSize": 0.05,
      "fontStyle": "normal",
      "fontWeight": "regular",
    }
  }
  hideBounds={false}
  lineSpacing={1}
  localPosition={
    Array [
      1,
      1,
      1,
    ]
  }
  localRotation={
    Array [
      1,
      1,
      1,
      1,
    ]
  }
  localScale={
    Array [
      1,
      1,
      1,
    ]
  }
  localTransform={
    Array [
      1,
      1,
      1,
      1,
      2,
      2,
      2,
      2,
      3,
      3,
      3,
      3,
      4,
      4,
      4,
      4,
    ]
  }
  skipRaycast={false}
  textAlignment="center"
  textColor="white"
  textSize={0.05}
  visible={true}
>
  MagicScript Components
</text>
`;

exports[`Text component with fontSize == 0.05 1`] = `
<text
  alignment="center-center"
  allCaps={false}
  boundsSize={
    Object {
      "boundsSize": Array [
        0,
        0,
      ],
      "wrap": false,
    }
  }
  charSpacing={0}
  debug={false}
  enabled={true}
  fontParameters={
    Object {
      "allCaps": false,
      "fontSize": 0.05,
      "fontStyle": "normal",
      "fontWeight": "regular",
    }
  }
  hideBounds={false}
  lineSpacing={1}
  localPosition={
    Array [
      0,
      0,
      0,
    ]
  }
  localRotation={
    Array [
      0,
      0,
      0,
      1,
    ]
  }
  localScale={
    Array [
      1,
      1,
      1,
    ]
  }
  localTransform={
    Array [
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      0,
    ]
  }
  skipRaycast={false}
  textAlignment="left"
  textColor="white"
  textSize={0.05}
  visible={true}
>
  MagicScript Components
</text>
`;

exports[`Video component 1`] = `
<video
  path="resources/video.mp4"
  subtitlePath="resources/subtitles.txt"
/>
`;
